crates/wasi-provider/src/lib.rs

```diff
@@ -188,12 +187,13 @@ impl Provider for WasiProvider {
                 pod_name,
                 container_name,
             })?;
-
-        Ok((&mut handle.0)
-            .lines()
-            .map(|l| l.unwrap_or_else(|_| "<error while reading line>".to_string()))
-            .collect::<Vec<String>>()
-            .await)
+        let mut output = Vec::new();
+        (&mut handle.0).read_to_end(&mut output).await?;
+        // Reset the seek location for the next call to read from the file
+        // NOTE: This is a little janky, but the Tokio BufReader does not
```

Comment:
I don't think this is very janky... What do we want to say with this comment?

Reply:
I don't like pulling out an inner mutable ref to a read object just to seek, especially when the standard lib equivalent has the ability to do this without the inner mutable ref

Reply:
True. Not sure it warrants a comment, but I agree